### ISSUE TITLE
Triage important tasks

### ISSUE THREAD

**Issue by iiegor** [iiegor/xat-server#7]
**:arrow_up: xat-server@0.1.0**

---
- [x] Correct load and handle of the user powers.
  - [x] Calculate power values _(d4, d5, d6..)_
  - [x] Create user powers attributes from user packet _(p0, p1, p2..)_
- [x] Correct user authentication ([workers/user.coffee](https://github.com/iiegor/xat-server/blob/master/src/workers/user.coffee)).
  - [x] Normal user authentication
  - [x] Guest user authentication
- [ ] Verify user identity when sending commands ([#L17](https://github.com/iiegor/xat-server/blob/master/src/workers/commander.coffee#L17)).
- [ ] Select only needed data ([#L5](https://github.com/iiegor/xat-server/blob/master/src/workers/profile.coffee#L5)).
- [ ] Add a task to compile coffee-script code for production.
- [x] ~~Support for multiple queries using the same connection ([#L30](https://github.com/iiegor/xat-server/blob/master/src/services/database.coffee#L30)).~~
- [x] Ping the database before sending **application:started** event ([src/bootstrap.coffee](https://github.com/iiegor/xat-server/blob/master/src/bootstrap.coffee)).
- [x] Limit messages; Fix bug using order desc and limit.
- [x] ⚠ Implement a packet builder instead of hardcoding the packets structure.
- [x] Bind services with the main handler class and avoid passing handler class each time.

**Comment by huumanoid** [comment]
Let's discuss.
What is left to do with Correct user authentication?

NOTE:

``` coffee
class Test
  u: { m: 0 }
```

it's unsafe way to initialize bar. in this way, all instances of Test works with same copy of u

``` coffee
a = new Test
a.u.m++
b = new Test
console.log(b.u,m)
```

guess what? ;)
this note related to iiegor/xat-server@9a476e6ac8ce741a644b5562e2ccb1a87170c173 
It seems that it's not tempfix, it is the safe way to initialize 'user' field.

**Comment by iiegor** [comment]
@HuuMaNoID It seems that only remains to fix the authentication for guest users and probably complete some missing user packets, regarding that task.

wow, you are completely right about the initialization of the variables :fearful:

_Non-related: I added you as a collaborator so this way we can make more easier the development. Only push directly to the repo very important and small changes, for the others create pull requests to discuss the changes._

**Comment by huumanoid** [comment]
**Verify user identity when sending commands**
This task requires user rank implementation, isn't it?

**Select only needed data**
How do you see it? Maybe we may pass required fields to getById function?

**Comment by iiegor** [comment]
@HuuMaNoID Yes, the user identity verification requires of the user rank. Regarding the second question, I think your idea it's perfect ;)
